app/src/main/java/com/rockthevote/grommet/ui/registration/RegistrationViewModel.kt

```diff
@@ -3,61 +3,138 @@ package com.rockthevote.grommet.ui.registration
+    // TODO this should not make it into merge, it's a placeholder for data I don't know the origin of yet
+    private val unknownDataSource: UnknownDataSource = UnknownDataSource(true, true, true, "temp", "temp", GeoLocation(1.0, 1.0), "temp")
+) : ViewModel() {
     private val _registrationData = MutableLiveData(RegistrationData())
     val registrationData: LiveData<RegistrationData> = _registrationData
 
+    private val _registrationState: MutableLiveData<RegistrationState> = MutableLiveData(RegistrationState.Init)
+    val registrationState: LiveData<RegistrationState> = _registrationState
+
+    private val currentData
+        get() = _registrationData.value ?: RegistrationData()
+
     fun storeNewRegistrantData(data: NewRegistrantData) {
-        val newData = _registrationData.value?.copy(
+        val newData = currentData.copy(
             newRegistrantData = data
         )
 
         updateData(newData)
     }
 
     fun storeAddressData(data: PersonalInfoData) {
-        val newData = _registrationData.value?.copy(
+        val newData = currentData.copy(
             addressData = data
         )
 
         updateData(newData)
     }
 
     fun storeAdditionalInfoData(data: AdditionalInfoData) {
-        val newData = _registrationData.value?.copy(
+        val newData = currentData.copy(
             additionalInfoData = data
         )
 
         updateData(newData)
     }
 
     fun storeAssistanceData(data: AssistanceData) {
-        val newData = _registrationData.value?.copy(
+        val newData = currentData.copy(
             assistanceData = data
         )
 
         updateData(newData)
     }
 
-    fun storeReviewData(data: ReviewData) {
-        val newData = _registrationData.value?.copy(
+    private fun storeReviewData(data: ReviewData) {
+        val newData = currentData.copy(
             reviewData = data
         )
 
-        updateData(newData)
+        updateData(newData, true)
     }
 
-    private fun updateData(data: RegistrationData?) {
+    fun completeRegistration(data: ReviewData) {
+        storeReviewData(data)
+
+        runCatching {
+            val transformer = RegistrationDataTransformer(currentData, partnerInformation)
+            val requestData = transformer.transform(unknownDataSource)
+
+            /* TODO Either inject an adapter, make this an asnc operation, or just pass the data
+                object to whatever class handles storing this to offload the responsibility
+             */
+            val adapter = Moshi.Builder()
+                .add(KotlinJsonAdapterFactory())
+                .build()
+                .adapter(RockyRequest::class.java)
+
+            val rockyRequestJson = adapter.toJson(requestData)
+
+            Timber.d("Storing RockyRequest JSON %s", rockyRequestJson)
+
+            // TODO Send request data to DB
+        }.onSuccess {
+            updateState(RegistrationState.Complete)
```

Comment:
I think by definition if it's stored that it's "complete", no?

Reply:
This just emits the state to the view so it can act on the Completion and finish `ReviewAndConfirmFragment` (showing the complete dialog in `completeForm`)